### ISSUE TITLE
install -g /root/fuzzingandroid/aut_apk/aut.apk

### ISSUE THREAD

**Issue by nicoostendorf** [DroidTest/TimeMachine#12]
Hello, 
i could run Time Machine with the two example Apps. I tried to run with an app from the AndroTest App set. 
I first wanted it to try with Ella. After i get Ella to run on my device, i instrumented the timer and the translate App. 
I Instrumented the bin/TimerActivity-debug.apk and bin/TranslateAcitvity-debug.apk with the edited file for Ella from the solved issue. 

When i now try to run TimeMachine it hangs up in step:
```
+ echo 'install app under test'
install app under test
+ adb -s 127.0.0.1:6000 install -g /root/fuzzingandroid/aut_apk/aut.apk
```

I ran TimeMachine with the following command:

`./exec-single-app.bash ../two_apps_under_test/timer/ 0 droidtest/timemachine:1.0 1800 ../timer_output`

I have no idea why it does that could you please help me?

My instrumented apps and Covids are:
[timer.zip](https://github.com/DroidTest/TimeMachine/files/8755670/timer.zip)
[Translate.zip](https://github.com/DroidTest/TimeMachine/files/8755673/Translate.zip)

Thanks for your help!

**Comment by SioYooo** [comment]
So sorry, some problem happed so the previous answers for the issue are not successfully posted.

In my understanding, these step

``` shell
+ echo 'install app under test'
install app under test
+ adb -s 127.0.0.1:6000 install -g /root/fuzzingandroid/aut_apk/aut.apk
```

is just to install the apk where you chose onto the Android VM created by TimeMachine, so I think if you are stuck at this point, I think you should check if your APK can be successfully installed on the VM with the same environment.

My log is like this: 

```shell
+ echo 'install app under test'
install app under test
+ adb -s 127.0.0.1:6000 install -g /root/fuzzingandroid/aut_apk/aut.apk
Success
+ sleep 5
+ echo 'shut down VM after installing app'
shut down VM after installing app
+ VBoxManage controlvm Android7_1 poweroff
0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
+ cd FuzzerEngine/fuzzerengine
+ Mode=gui
+ ((  1 == 1  ))
```

And BTW, I would like to know if your apk is building in the recent Android version? i.e. after Android 8.0, since I am facing difficulties with instrumenting applications using Ella for more recent apks.

I have tried instrument apks that I downloaded from the internet which were released a few months ago, they are not protected, however, a log of bugs occurred while instrumenting.

Please let me know if you can help me with this.

All the best.

**Comment by mboehme** [comment]
Just chiming in to get some more information. I think @zhendong2050 (@DroidTest) might be better suited to answer these questions.

@nicoostendorf What is the problem you experience after running `adb -s 127.0.0.1:6000 install -g /root/fuzzingandroid/aut_apk/aut.apk`

@SioYooo Is this a different question? It seems you install command returns "Success"? If it is a different question, can you just leave your answer here and open a new Issue for your follow-up question, also to be more concrete on the "lot of bugs"?

**Comment by SioYooo** [comment]
> Just chiming in to get some more information. I think @zhendong2050 (@DroidTest) might be better suited to answer these questions.
> 
> @nicoostendorf What is the problem you experience after running `adb -s 127.0.0.1:6000 install -g /root/fuzzingandroid/aut_apk/aut.apk`
> 
> @SioYooo Is this a different question? It seems you install command returns "Success"? If it is a different question, can you just leave your answer here and open a new Issue for your follow-up question, also to be more concrete on the "lot of bugs"?

Yes, in my testing, the system returns install success after the installation, so I suggest @nicoostendorf test if the instrumented apk can be installed on another device to test what is the problem.

I am currently collecting bugs when I am trying to instrument apks using Ella that is modified with the Ella.java file provided by zhendong.

I will post another issue shortly.

Thanks for answering.

**Comment by mboehme** [comment]
Thanks @SioYooo! Did it resolve your issue @nicoostendorf?

**Comment by nicoostendorf** [comment]
It doesnt solve my Problem..

The System stucks at the install step.

I am not able to install the instrumented apks by my self, so i think the Problem is not in TimeMachine..
Maybe something went wrong with ella..


@SioYooo when i am trying to instrument more recent apps it gave me a few errors... I asked @zhendong2050 and he said Ella and Emma are out of date... And they are working on an alternative...

If anyone of you is able to instrument apks with ella i would be happy if you could help me with this. 😵‍💫

**Comment by tingsu** [comment]
I would like to give some comments. Hope it may help. @nicoostendorf 

I used Ella before when building my app testing tool. My feeling is Ella was out of date and not actively maintained anymore. If you have to use Ella, you may have to inspect Ella's source code and modify it appropriately.

If you target open-source apps, you can use Jacoco (which is officially supported by Android). In fact, I successfully integrated Jacoco into TimeMachine to fuzz open-source apps. This is [my code](https://github.com/the-themis-benchmarks/TimeMachine) for adapting TimeMachine.  I did this adaption for evaluating the bug finding ablilities of many existing GUI fuzzing tools (see [Themis](https://github.com/the-themis-benchmarks/home)).

**Comment by zhendong2050** [comment]
Thanks @mboehme @tingsu for answers. 

@nicoostendorf A new version of TimeMachine was just released. You can follow README to instrument apps with Jacoco.

**Comment by zhendong2050** [comment]
Thanks @mboehme @tingsu for answers.

@nicoostendorf A new version of TimeMachine was just released. You can follow README to instrument apps with Jacoco.
